core/src/main/java/org/infinispan/util/function/RemovableFunction.java

```diff
@@ -0,0 +1,14 @@
+package org.infinispan.util.function;
+
+import java.util.function.Function;
+
+/**
+ * Placeholder interface used to describe a function that can be used as is for removal.  This is useful for
+ * allowing an iterator to use removal.
+ * <p>
+ * Users shouldn't ever need to use this interface directly
```

Comment:
Yeah, it doesn't make a lot of sense for the users to implement it.

The naming bothers me a bit, but I can't think of a better one... So instead, I'm wondering if you can just assume that in a compatibility/storeAsBinary cache the first mapping will be a conversion? It's not like an iterator without the conversion function could support `remove()` anyway, right?

Off-topic, what's the use case for going from a stream to a mutable iterator? I noticed the JDK streams don't support removal.